docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,455 @@
+| Option B.1: Create separate semantic FluentProvider             | • Separates Fluent Semantic Tokens (FSTs) from FluentProvider and Theme.<br>• Optional layering so partners only pull it in when needed. | • Requires extra layer<br>• May not be as intuitive as just using the existing provider already in applications<br>• May not provide much value since the components still need to have some level of awareness of the group and component tokens. |
+| Option B.2: Create separate semantic FluentProvider and package | • Same pros as above but with added flexibility around packaging and versioning                                                          | See above                                                                                                                                                                                                                                          |
+
+## Update components to optionally consume expanded tokens
+
+Once exposed in the theme, the new tokens could be accessed in existing components simply by updating existing references to alias tokens with var(), passing the more precise token as the first argument and retaining original alias token as the second.
+
+For example, instead of:
+
+```javascript
+const useRootBaseClassName = makeStyles({ backgroundColor: 'var(--colorNeutralForeground1)' });
+```
+
+We would use this:
+
+```javascript
+const useRootBaseClassName = makeStyles({
+  backgroundColor: 'var(--buttonBackgroundColorRest, var(--colorNeutralForeground1))',
+});
+```
+
+This leverages `var()`’s fallback feature to read the new token’s value first only if defined in the theme, while falling back on the existing alias token if not defined.
+
+To ensure backwards compatibility with existing themes, all v9 components consuming the new tokens should **always** use alias token fallbacks.
+
+## Performance
+
+Given [previously observed performance issues](https://github.com/microsoft/fluentui/blob/6e6a1bf624e5a682b3607d918793d6f0eeb6b12a/rfcs/react-components/convergence/theme-shape.md#performance-analysis) when injecting a class with many CSS variables that referenced other CSS variables, we wanted to be sensitive to bloating the theme object with additional tokens. We noticed that performance penalties in the original tests were not concentrated at variable lookup time, but instead due to the root-level nature of the custom properties: each of the FluentProviders injected tokens that were associated with DOM elements, causing more reflows than would be expected with more scoped properties.
+
+Our hypothesis is that more granularly scoped tokens tied to fewer DOM nodes would not incur the same performance penalties, although of course this needs to be tested.
```

Comment:
Can you please connect with Edge folks and ensure that it's true?

Reply:
For sure, connecting with Jane and Chris (who are talking with Edge folks)